libethereum/BlockChainSync.cpp

```diff
@@ -217,7 +217,7 @@ bool BlockChainSync::requestDaoForkBlockHeader(NodeID const& _peerID)
         return false;
 
     m_daoChallengedPeers.insert(_peerID);
-    m_host.peer(_peerID).requestBlockHeaders(daoHardfork, 1, 0, false);
+    m_host.peer(_peerID).requestBlockHeaders(static_cast<unsigned>(daoHardfork), 1, 0, false);
```

Comment:
After `daoHardfork` type was changed to `u256` in https://github.com/ethereum/aleth/pull/5539 overload resolution preferred `requestBlockHeader(h256  const&, ...)` instead of `requestBlockHeader(unsigned, ...)`, sync with the main net was not working

Reply:
Why do we need the other overload?

Reply:
eth wire protocol contains request for headers both by number and by hash. We request the latest known block header by hash, when we begin to sync from the peer. Because we receive only this hash in the Status message.

Reply:
Strangely Status message spec includes block number,too https://github.com/ethereum/devp2p/blob/master/caps/eth.md#status-0x00
But quick experiment shows that no one actually sends block number, all Status messages contain only hash. Aleth also doesn't send it.